packages/main/src/MultiComboBox.js

```diff
@@ -1303,6 +1311,12 @@ class MultiComboBox extends UI5Element {
 			item._getRealDomRef = () => this.allItemsPopover.querySelector(`*[data-ui5-stable=${item.stableDomRef}]`);
 		});
 
+		const slottedIconsCount = this.icon ? this.icon.length : 0;
```

Comment:
Here also you can extract these thins in a function to return the total count and directly pass it to setProperty. 
The onBeforeRendering already seems very big, so extracting actions would make it more readable.